bin/cancel_squad_testjobs.py

```diff
@@ -1,74 +1,84 @@
-""",
-    )
-    parser.add_argument(
-        "--project-slug",
-        dest="project_slug",
-        required=True,
-        help="slug of the project which test jobs will be canceled",
-    )
-    parser.add_argument(
-        "--squad-url",
-        dest="squad_url",
-        required=True,
-        help="URL of SQUAD instance in form https://example.com",
-    )
-    parser.add_argument(
-        "--build-version",
-        dest="build_version",
-        required=True,
-        help="Version of the build from which testjobs will be canceled",
-    )
+        Given something like the following:
+            url="https://qa-reports.linaro.org"
+            project="linux-stable-rc-4.9-oe"
+            build="v4.9.162-94-g0384d1b03fc9"
 
-    args = parser.parse_args()
-    params = {"slug": args.project_slug}
-    base_url = args.squad_url + "/api/projects/"
+        Discover and cancel all lava jobs that are still running.
+
+        Note this doesn't handle duplicate project names well..
+    """
+
+    base_url = squad_client.urljoiner(url, "/api/projects/")
 
-    project = get_objects(base_url, True, params)
-    build_list = get_objects(project["builds"], {"version": args.build_version})
+    params = {"slug": project}
+    project = squad_client.get_objects(base_url, True, params)[0]
+    build_list = squad_client.get_objects(project["builds"], {"version": build_version})
+    identity_argument = ""
+    if identity:
+        identity_argument = "-i {}".format(identity)
     for build in build_list:
-        if build["version"] != args.build_version:
+        if build["version"] != build_version:
             # double check. but also, version filter is broken presently
             continue
 
-        testjobs = get_objects(build["testjobs"])
+        testjobs = squad_client.get_objects(build["testjobs"])
         for testjob in testjobs:
             if testjob["job_status"] != "Submitted":
                 print(
                     "Skipping: %s; status: %s"
                     % (testjob["job_id"], testjob["job_status"])
                 )
                 continue
-            backend = get_objects(testjob["backend"])
-            print(
-                "Canceling: %s/scheduler/job/%s"
-                % (backend["url"].replace("/RPC2/", ""), testjob["job_id"])
+            backend = squad_client.get_objects(testjob["backend"])
+            print("Canceling: %s" % (testjob["job_id"]))
+
+            cmd = "lavacli {} jobs cancel {}".format(
+                identity_argument, testjob["job_id"]
             )
-            a = Auth(backend["url"])
-            l = LAVA(backend["url"], a.username, a.token)
-            c_results = l.proxy.scheduler.cancel_job(testjob["job_id"])
-            pprint(c_results)
+            print(cmd)
+            subprocess.check_call(cmd, shell=True)
```

Comment:
IMHO there should be some check for existence of lavacli. Otherwise this call might throw exception. From the docs "If the return code was zero then return, otherwise raise CalledProcessError"

Reply:
I was about to fix it, but the output is actually very clear (below) and I think wrapping it would only serve to hide meaningful information.

```
Canceling: 640843
lavacli  jobs cancel 640843
/bin/sh: lavacli: command not found
Traceback (most recent call last):
  File "/home/drue/bin/cancel_squad_testjobs.py", line 87, in <module>
    cancel_lava_jobs(url, project, build_version, args.identity)
  File "/home/drue/bin/cancel_squad_testjobs.py", line 57, in cancel_lava_jobs
    subprocess.check_call(cmd, shell=True)
  File "/usr/lib64/python3.7/subprocess.py", line 347, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'lavacli  jobs cancel 640843' returned non-zero exit status 127.
drue@xps:~$ 
```